app/tests/integration/nimbus/conftest.py

```diff
@@ -94,6 +94,23 @@ def firefox_options(firefox_options):
     return firefox_options
 
 
+@pytest.fixture
+def selenium(selenium, experiment_name, kinto_client, base_url, slugify):
+    yield selenium
+
+    from nimbus.pages.experimenter.summary import SummaryPage
+
+    experiment_slug = str(slugify(experiment_name))
+    try:
+        summary = SummaryPage(selenium, urljoin(base_url, experiment_slug)).open()
+        summary.end_and_approve()
+        kinto_client.approve()
+        summary = SummaryPage(selenium, urljoin(base_url, experiment_slug)).open()
+        summary.wait_for_complete_status()
+    except Exception:
+        pass
```

Comment:
Oh so this will run on every experiment end, the name `selenium` means it'll override the existing selenium fixture so it'll automatically be included in every test, and it'll pull the experiment_name out of the other fixture so it'll arleady be the same for each test, and then we just explicitly end it....  So that tacks on the time to do the end on top of the length of each test run but cleans the experiment up at the end....

Yeah that's cool, let's try it and see if it helps.